internal/distro/distro.go

```diff
@@ -35,7 +35,7 @@ type Distro interface {
 	// Returns the canonical filename and MIME type for a given output
 	// format. `outputFormat` must be one returned by
 	// TODO: remove when redundant
-	FilenameFromType(outputFormat string) (string, string, error)
+	FilenameFromType(arch, outputFormat string) (string, string, error)
```

Comment:
Hm, if this function takes an `arch` argument, then it means we have all the information to resolve the real `ImageType` object in the caller, and imply call `Filename()` on that. There was a reason I thought we couldn't, but seeing your patch, I think it should be ok. See below.